src/metabase/routes.clj

```diff
@@ -10,7 +10,9 @@
 
 (defn- index [_]
   (-> (if (@(resolve 'metabase.core/initialized?))
-        (stencil/render-string (slurp (io/resource "frontend_client/index.html"))
+        (stencil/render-string (slurp (or (io/resource "frontend_client/index.html")
+                                          ;; Occasionally people try to launch from 'lein ring server' and move stuff around. In that case give them a useful error message.
+                                          (throw (Exception. "Cannot find './resources/frontend_client/index.html'."))))
```

Comment:
Can we add one additional line in the exception saying something like, "Have you built the Metabase frontend?"